### DISCUSSION TITLE
Chapter 09. 정규화

### DISCUSSION THREAD

**Discussion by leegyeongwhan** [CodeSquad-2023-BE-Study/DB-Study#13] [완료]
책을 읽으면서 궁금한 점을 남겨주세요.

**Comment by leegyeongwhan** [comment]
개인적으로 BCNF, 3정규형이 이해가 잘 안가 https://www.youtube.com/watch?v=5QhkZkrqFL4&list=PLcXyemr8ZeoREWGhhZi5FZs6cvymjIBVe&index=24
유튜브 보면서 공부했는데 괜찮네요

결국 정규화를 하기전에 중요한 것은 함수 종속성이다라는 생각이 들었습니다.
각 attribute 간의 종속성에 대해 생각하는 것을 정규화의 시작으로 진행해야겠네요. 생각보다 근데 우리가 정규화를 배우지 않아도
나름 정규화를 하고있었다?? 라는 생각도 하게 됐습니다.

**Reply by leegyeongwhan** [reply]
https://rebro.kr/160 이블로그도 괜찮네요

**Comment by jinny-l** [comment]
#331~332
삽입 이상의 예시가 잘 이해가 안되네요.
신규 회원이 있고, 이벤트를 참여하지 않았는데 왜 이벤트 참여 테이블에 데이터를 삽입하려고 하는지?

**Reply by jinny-l** [reply]
아 정규화가 안되어 있어서 회원 테이블이랑 이벤트 테이블이 같이 있는 경우였군요

**Comment by new-pow** [comment]
함수적 종속성과 릴레이션, 정규화의 관계를 잘 몰랐는데 잘 정리할 수 있었습니다.
> 이상 현상이 발생하지 않도록 하려면, 관련있는 속성들로만 릴레이션을 구성해야하는데 이를 위해 필요한 것이 정규화이다. (중략) 일반적으로 릴레이션에 함수적 종속성이 하나 존재하도록 정규화를 통해 릴레이션을 분해한다.

**Comment by new-pow** [comment]
365p 25번 문제 대체 뭐죠...?

**Reply by new-pow** [reply]
![image](https://github.com/CodeSquad-2023-BE-Study/DB-Study/assets/103120173/d5d8ccfa-7af1-4f57-9dd9-7124a59a52db)

**Reply by jinny-l** [reply]
?-?

**Reply by jinny-l** [reply]
B → A
C → (A, B, D)
D → A
{A, C} → (B, D)
{B, C} → (A, D)
{B, D} → (A, C)
{C, D} → (A, B)

**Reply by jinny-l** [reply]
{A, B, C} → D
{A, B, D} → C
{A, C, D} → B
{B, C, D} → A

**Reply by jaea-kim** [reply]
함수 종속성이 쉽게 말하면 x 값이 y를 결정하니까 같은 x를 넣었을 때 같은 y가 나오는 조합을 다 적으라는 것 같다..는 .. 듀이의 뇌피셜..

**Comment by new-pow** [comment]
BCNF는 무엇인가

**Reply by new-pow** [reply]
- 모든 결정자가 후보키여야 한다.

**Reply by new-pow** [reply]
![image](https://github.com/CodeSquad-2023-BE-Study/DB-Study/assets/103120173/7d3efbb8-9e95-4020-9eba-7b63763bba5c)
